src/action/stat.rs

```diff
@@ -120,14 +106,19 @@ pub fn handle<S: Session>(session: &mut S, request: Request) -> session::Result<
 
 #[cfg(target_family = "unix")]
 fn get_ext_attrs(path: &Path) -> Vec<ExtAttr> {
+    use std::os::unix::ffi::OsStringExt;
     let xattrs = xattr::list(path).unwrap();
 
     let mut result = vec![];
     for attr in xattrs {
-        result.push(ExtAttr {
-            name: Some(attr.to_str().unwrap().as_bytes().to_vec()),
-            value: xattr::get(path, attr).unwrap(),
-        });
+        match xattr::get(path, &attr) {
+            Ok(attr_value) => result.push(ExtAttr {
+                name: attr.into_vec(),
+                value: attr_value.unwrap_or_default(),
+            }),
+
+            Err(_) => ()
```

Comment:
I would at least log this error.

Reply:
[Fixed](https://github.com/google/rrg/pull/34/commits/9277f329f74fe9fc09b42973594738928817d14e)